project_task_stage_change_restriction/models/project_task_stage.py

```
@@ -0,0 +1,36 @@
+
+    assigned_only = fields.Boolean()
+    project_manager_only = fields.Boolean(string="Project Manager")
+    allowed_group_ids = fields.Many2many(
+        "res.groups",
+        "project_task_stage_group_rel",
+        "stage_id",
+        "group_id",
+        string="Group Members",
+    )
+
+    def has_restrictions(self):
```

Ruff (0.11.9)

Missing return type annotation for public function `has_restrictions`

(ANN201)